routing_common/routing_common_tests/vehicle_model_for_country_test.cpp

```diff
@@ -0,0 +1,155 @@
+#include "testing/testing.hpp"
+
+#include "routing_common/bicycle_model.hpp"
+#include "routing_common/car_model.hpp"
+#include "routing_common/pedestrian_model.hpp"
+
+#include "indexer/classificator.hpp"
+#include "indexer/classificator_loader.hpp"
+
+#include <memory>
+#include <string>
+
+using namespace routing;
+using namespace std;
+
+namespace
+{
+class  VehicleModelTest
+{
+public:
+  VehicleModelTest() { classificator::Load(); }
+};
+
+string GetRegionParent(string const & id)
+{
+  if (id == "Moscow")
+    return "Russian Federation";
+  if (id == "Munich")
+    return "Bavaria";
+  if (id == "Bavaria")
+    return "Germany";
+  if (id == "San Francisco")
+    return "California";
+  if (id == "California")
+    return "United States of America";
+  return "";
+}
+
+template <typename VehicleModelType, typename VehicleModelFactoryType>
+void TestVehicleModelDefault()
+{
+  auto defaultVehicleModel = make_shared<VehicleModelType>();
+
+  VehicleModelFactoryType vehicleModelFactory = VehicleModelFactoryType(GetRegionParent);
+
+  // To not add virtual EqualsForTests to VehicleModelInterface, CrossCountryVehicleModel etc
+  shared_ptr<VehicleModelType> defaultVehicleModelForCountry = static_pointer_cast<VehicleModelType>(
+      vehicleModelFactory.GetVehicleModelForCountry("Nonexistent Country Name"));
+  TEST(defaultVehicleModel->EqualsForTests(*defaultVehicleModelForCountry),
+       ("Vehicle model for nonexistent counry is not equal to default."));
+}
+
+// DirectParent and IndirectParent tests require tested countries to have nondefault restriction
+// If selected countries will change they restrictions to default select other countries for tests
```

Comment:
s/they/their/
s/to default/to default,/

Reply:
done